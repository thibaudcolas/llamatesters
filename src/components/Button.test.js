import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    it('renders', () => {
        expect(shallow(<Button />)).toMatchInlineSnapshot(`
            <button
              className="button button-rounded  "
              type="button"
            />
        `);
    });

    it('#iconBefore', () => {
        expect(
            mount(<Button iconBefore="rocket">Test</Button>),
        ).toMatchSnapshot();
    });

    it.todo('#iconAfter');

    it.skip('#icon', () => {
        expect(
            shallow(<Button icon="rocket" />)
                .find('Icon')
                .props().name,
        ).toBe('potato');
    });

    it.skip('can be clicked', () => {
        const onClick = jest.fn();

        onClick();
        // const wrapper = shallow(<Button onClick={onClick}>Test</Button>);
        // wrapper.find('button').simulate('click');

        expect(onClick).toHaveBeenCalled();
    });
});
