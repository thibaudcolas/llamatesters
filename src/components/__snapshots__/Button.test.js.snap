// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button #iconBefore 1`] = `
<Button
  className={null}
  href={null}
  icon={null}
  iconAfter={null}
  iconBefore="rocket"
  theme="default"
>
  <button
    className="button button-rounded  "
    type="button"
  >
    <Icon
      className="button__icon button__icon--before"
      name="rocket"
    >
      <svg
        aria-hidden="true"
        className="icon icon--rocket button__icon button__icon--before"
      >
        <use
          xlinkHref="#rocket"
        />
      </svg>
    </Icon>
    Test
  </button>
</Button>
`;
